src/extension.ts

```diff
@@ -38,27 +39,74 @@ export function activate(context: vscode.ExtensionContext) {
   // listen for configuration changes
   context.subscriptions.push(
     vscode.workspace.onDidChangeConfiguration((e) => {
-      const configChanged = e.affectsConfiguration('vimFindHighlight.charPrimaryColor') || e.affectsConfiguration('vimFindHighlight.charSecondaryColor') || e.affectsConfiguration('vimFindHighlight.charFontWeight') || e.affectsConfiguration('vimFindHighlight.enableUnderline');
+      const configChanged =
+        e.affectsConfiguration("vimFindHighlight.charPrimaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charSecondaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charFontWeight") ||
+        e.affectsConfiguration("vimFindHighlight.enableUnderline") ||
+        e.affectsConfiguration("vimFindHighlight.enableAutoHighlight");
 
       if (configChanged) {
         configureDecoration();
+        autoHighlight = configureAutoHighlight();
       }
     })
-  )
+  );
 
-  // when activating the extension. read the user settings file first
+  // when activating the extension. read the user settings file first and register the commands
   configureDecoration();
+  registerCommands(context);
 }
 
 const configureDecoration = () => {
   updateDecorationConfig();
   disposeCharDecoration();
 };
 
+const configureAutoHighlight = (): boolean => {
+  const settings = vscode.workspace.getConfiguration();
+  const autoHighlight = settings.get(
+    "vimFindHighlight.enableAutoHighlight"
+  ) as boolean;
+  vscode.window.showInformationMessage(autoHighlight.toString());
```

Comment:
I think we can remove this. Most people will probably configure this value through the settings such that there is no need to see it pop up everytime they open vscode.

Reply:
Lmao this was for my own testing! Will remove